modules/related-posts/jetpack-related-posts.php

```diff
@@ -591,11 +591,13 @@ public function print_setting_head() {
 	public function get_for_post_id( $post_id, array $args ) {
 		$options = $this->get_options();
 
-		if ( ! empty( $args['size'] ) )
+		if ( ! empty( $args['size'] ) ) {
 			$options['size'] = $args['size'];
+		}
 
-		if ( ! $options['enabled'] || 0 == (int)$post_id || empty( $options['size'] ) )
+		if ( ! $options['enabled'] || 0 == (int)$post_id || empty( $options['size'] ) || get_post_status( $post_id) !== 'publish' ) {
```

Comment:
You may want to use a Yoda condition here.

Reply:
I didn't think we needed a Yoda since we're not dealing with a variable?

Reply:
That works for me. I would personally use a Yoda for consistency, but it's really nitpicking. Marking as ready to merge. 👍